.jenkins/pipelines/code-coverage.Jenkinsfile

```diff
@@ -5,79 +5,325 @@
 
 pipeline {
     agent {
-        label 'Jenkins-Shared-DC2'
+        label 'ACC-v3-2004'
     }
     options {
-        timeout(time: 300, unit: 'MINUTES')
+        timeout(time: 600, unit: 'MINUTES')
         timestamps ()
     }
     parameters {
         string(name: "REPOSITORY", defaultValue: "deislabs/mystikos")
         string(name: "BRANCH", defaultValue: "main", description: "Branch to build")
         choice(name: "REGION", choices:['useast', 'canadacentral'], description: "Azure region for the SQL solutions test")
         string(name: "PULL_REQUEST_ID", defaultValue: "", description: "If you are building a pull request, enter the pull request ID number here. (ex. 789)")
+        string(name: "COMMIT_SYNC", defaultValue: "", description: "optional - used to sync outputs of parallel jobs")
+        string(name: "PACKAGE_NAME", defaultValue: "", description: "optional - release package to install (do not include extension)")
+        choice(name: "PACKAGE_EXTENSION", choices:['tar.gz', 'deb'], description: "Extension of package given in PACKAGE_NAME")
     }
     environment {
         TEST_CONFIG = 'Code Coverage'
+        UBUNTU_VERSION = '18.04'
+        VM_GENERATION = 'v3'
+        MYST_SCRIPTS =      "${WORKSPACE}/scripts"
+        JENKINS_SCRIPTS =   "${WORKSPACE}/.jenkins/scripts"
+        MYST_NIGHTLY_TEST = 1
+        MYST_ENABLE_GCOV =  1
+        LCOV_FILE =         "lcov-${GIT_COMMIT[0..7]}.info"
+        LCOV_DIR="mystikos-cc-${GIT_COMMIT[0..7]}"
+        BUILD_INFO =      "${BUILD_URL == '' ? 'N/A' : BUILD_URL}"
+        PACKAGE_INSTALL =   "${UBUNTU_VERSION == '20.04' ? 'Ubuntu-2004' : 'Ubuntu-1804'}_${PACKAGE_NAME}.${PACKAGE_EXTENSION}"
+        BUILD_USER = sh(
+            returnStdout: true,
+            script: 'echo \${USER}'
+        )
     }
     stages {
-        stage('Run Code Coverage Tests') {
-            matrix {
-                axes {
-                    axis {
-                        name 'OS_VERSION'
-                        values '18.04'
+        stage("Cleanup files") {
+            steps {
+                sh "${JENKINS_SCRIPTS}/global/clean-temp.sh"
+            }
+        }
+        stage("Checkout Pull Request") {
+            when {
+                expression { params.PULL_REQUEST_ID != "" }
+            }
+            steps {
+                cleanWs()
+                checkout([$class: 'GitSCM',
+                    branches: [[name: "pr/${PULL_REQUEST_ID}"]],
+                    extensions: [],
+                    userRemoteConfigs: [[
+                        url: 'https://github.com/deislabs/mystikos',
+                        refspec: "+refs/pull/${PULL_REQUEST_ID}/merge:refs/remotes/origin/pr/${PULL_REQUEST_ID}"
+                    ]]
+                ])
+            }
+        }
+        stage('Verify commit sync') {
```

Comment:
Nit: you can remove this entire stage for Verify commit sync as this pipeline is running tests serially

Reply:
Removed